backend/views/book.js

```diff
@@ -286,24 +304,54 @@ module.exports = function(router) {
 			for (var i = 0; i < book.ratings.length; i++) {
 				if (book.ratings[i].id == userId) {
 					alreadyRatedByThisUser = true;
+					book.ratingsSum -= book.ratings[i]['rating'];
 					book.ratings[i]['rating'] = rating;	//Update the rating if already have rated
 					break;
 				}
 			}
 
 			if (!alreadyRatedByThisUser) {
+				book.ratingsSum += rating;
 				book.ratings.push({id: userId, rating: rating});
 				book.markModified('ratings');
+				book.avgRating = book.ratingsSum / book.ratings.length;
 				book.save();
-				res.json("Rated the book successfully");
 			}
 			else {
+				book.ratingsSum += rating;
 				book.markModified('ratings');
+				book.avgRating = book.ratingsSum / book.ratings.length;
 				book.save();
-				res.json("Updated the book rating successfully");
 			}
 
-		})
+			User.findOne({'_id': userId}, function findBookText(err, user) {
```

Comment:
@KarimElSheikh 
this function name is wrong
it is not findBookText
please either remove the name or name it with another name and document it